File: frontend/src/types/agent.ts
export type AgentState = 'idle' | 'running' | 'paused' | 'error' | 'stopped' | 'enabled';

export interface AgentStatus {
  is_running: boolean;
  is_paused: boolean;
  status: AgentState;
  current_task?: string | null;
  applications_made_today: number;
  applications_limit_today: number;
  applications_total: number;
  applications_limit_total: number;
  session_start_time?: string | null;
  last_activity_at?: string | null;
  errors_count: number;
  queue_size: number;
}

export interface AgentSettings {
  id: string;
  user_id: string;
  max_applications_per_day: number;
  cooldown_seconds: number;
  auto_apply: boolean;
  cover_letter_enabled: boolean;
  preferred_apply_time_start: string | null;
  preferred_apply_time_end: string | null;
  skip_easy_apply: boolean;
  require_salary_info: boolean;
  custom_answers: Record<string, unknown> | null;
  created_at: string;
  updated_at?: string;
}

export interface AgentLog {
  id: string;
  timestamp: string;
  level: 'info' | 'warning' | 'error' | 'success';
  message: string;
  details?: Record<string, unknown>;
}

export interface QueueItem {
  id: string;
  job_id: string;
  job_title: string;
  company: string;
  match_score: number;
  priority: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  added_at: string;
}

File: backend/app/schemas/agent.py
"""
Agent schemas for automation control.
"""

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AgentStatusResponse(BaseModel):
    """Schema for agent status."""
    model_config = ConfigDict(from_attributes=True)

    is_running: bool = False
    is_paused: bool = False
    status: str = "idle"  # idle, running, paused, error
    current_task: Optional[str] = None
    applications_made_today: int = 0
    applications_limit_today: int = 50
    applications_total: int = 0
    applications_limit_total: int = 10
    session_start_time: Optional[datetime] = None
    last_activity_at: Optional[datetime] = None
    errors_count: int = 0
    queue_size: int = 0


class AgentSettingsUpdate(BaseModel):
    """Schema for updating agent settings."""
    max_applications_per_day: Optional[int] = Field(None, ge=1, le=200)
    cooldown_seconds: Optional[int] = Field(None, ge=5, le=300)
    auto_apply: Optional[bool] = None
    preferred_apply_time_start: Optional[str] = None  # HH:MM format
    preferred_apply_time_end: Optional[str] = None
    skip_easy_apply: Optional[bool] = None
    require_salary_info: Optional[bool] = None
    cover_letter_enabled: Optional[bool] = None
    custom_answers: Optional[dict] = None


class AgentSettingsResponse(BaseModel):
    """Schema for agent settings response."""
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    max_applications_per_day: int = 50
    cooldown_seconds: int = 30
    auto_apply: bool = False
    preferred_apply_time_start: Optional[str] = None
    preferred_apply_time_end: Optional[str] = None
    skip_easy_apply: bool = False
    require_salary_info: bool = False
    cover_letter_enabled: bool = True
    custom_answers: Optional[dict] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class AgentLogResponse(BaseModel):
    """Schema for agent log entry."""
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    session_id: Optional[UUID] = None
    level: str = "info"  # info, warning, error, debug
    message: str
    details: Optional[dict] = None
    created_at: datetime
